staging/src/k8s.io/apiserver/pkg/admission/initializer/interfaces.go

```diff
@@ -89,3 +90,10 @@ type WantsSchemaResolver interface {
 	SetSchemaResolver(resolver resolver.SchemaResolver)
 	admission.InitializationValidator
 }
+
+// WantsResourceFilter defines a function which sets the ResourceFilter for
+// an admission plugin that needs it.
```

Comment:
Be very specific about which admission plugins will use this, possibly influencing the name of this function (`SetAdmissionPolicyResourceFilter`?)

Specifically, `{Validating,Mutating}AdmissionWebhook` plugins do *not*, and I would expect both `{Validating,Mutating}AdmissionPolicy` plugins will.

Do we really want all the power / flexibility of passing in something that gets actively invoked with full attributes on every admission request, or do we want this constrained to just an exclude-list of group/resources? The latter is easier to reason about, in my opinion.

Reply:
I agree injecting only the resource list makes more sense. The plugin can do whatever they want about the list. Hard-coding the behavior to ignoring is not ideal.